### DISCUSSION TITLE
Distribution Proposal: Truncated SkewNormal

### DISCUSSION THREAD

**Discussion by ericmjl** [pymc-devs/pymc3#5423] [Ideas]
@ricardoV94, I'm wondering what would it take to implement a Truncated SkewNormal distribution?

The main use case I have is to use it to construct unimodal posterior approximations.

SkewNormal is necessary if we want to model posteriors that are skewed off from a Gaussian-like shape.

Truncated is necessary if we want to approximate bound posteriors, i.e. things that are positive-only.

Do you have guidance here?

**Comment by ricardoV94** [comment]
We want to implement a general truncation solution that works for any distribution, similar to the general censored API we have now. The machinary would probably be implemented in Aeppl: https://github.com/aesara-devs/aeppl/issues/96

For an immediate solution all you need is the CDF of the skewnormal and you can implement it with an extra potential term and interval transform

**Comment by ricardoV94** [comment]
This should now be available with `pm.Truncated` assuming the logcdf is implemented for the `SkewNormal`
